iothub/service/src/Configurations/ConfigurationsClient.cs

```diff
@@ -69,15 +69,15 @@ public virtual async Task<Configuration> CreateAsync(Configuration configuration
             if (Logging.IsEnabled)
                 Logging.Enter(this, $"Adding configuration: {configuration?.Id}", nameof(CreateAsync));
 
-            try
+            Argument.AssertNotNull(configuration, nameof(configuration));
```

Comment:
Our desired order inside operational methods are:

1. Log enter
2. Check params
3. Check cancellation token cancelled
4. try/catch for operation with catch to log error and finally to log exit